main/streams/plain_wrapper.c

```diff
@@ -1114,11 +1120,14 @@ PHPAPI php_stream *_php_stream_fopen(const char *filename, const char *mode, zen
 	}
 
 	if (options & STREAM_ASSUME_REALPATH) {
-		strlcpy(realpath, filename, sizeof(realpath));
+		realpath_len = strlen(filename);
+		ZEND_ASSERT(realpath_len < sizeof(realpath) && "Assumed real path filename is too long");
+		strcpy(realpath, filename);
```

Comment:
This is an external API so there should be some prevention - you should at least use strncpy if you really need to replace this.

Reply:
you can keep the assert but not use only strcpy.

Reply:
strncpy does do padding, which is pointless. Actually, strlcpy here was fine IMO because with the new code we now do two traversals.

Reply:
yeah good point. Let's close this PR in that case...

Reply:
maybe just the top one makes sense. I don't really mind. This function is not even used from anywhere in the core - at least I didn't find any usage...

Reply:
there might still be external usage ofc. as it's public API.

Reply:
I dropped the first commit